### DISCUSSION TITLE
Board Game Geek scraper

### DISCUSSION THREAD

**Discussion by frank20a** [benjaminjonard/koillection#1065] [Show and tell]
I am sharing with everyone my scraping setup for Board Game Geek. I have checked that scraping a text type into an item number field converts correctly into a number. To use it, search for a game on BGG and look at its URL to find the game code. For example Terraforming Mars is [https://boardgamegeek.com/boardgame/167791/terraforming-mars](https://boardgamegeek.com/boardgame/167791/terraforming-mars) and the code is 167791. Then input the following URL to the scraper [https://boardgamegeek.com/xmlapi/boardgame/<game_code>](https://boardgamegeek.com/xmlapi/boardgame/167791) (Click the link for example).

```json
{
    "name": "Board Game Geek",
    "namePath": "#\/\/name[contains(@primary,'true')]\/text()#",
    "imagePath": "#\/\/image\/text()#",
    "urlPattern": "https:\/\/boardgamegeek.com\/xmlapi\/boardgame\/",
    "dataPaths": [
        {
            "name": "Released",
            "path": "#\/\/yearpublished\/text()#",
            "type": "text",
            "position": 1
        },
        {
            "name": "Players Minimum",
            "path": "#\/\/minplayers\/text()#",
            "type": "text",
            "position": 2
        },
        {
            "name": "Players Maximum",
            "path": "#\/\/maxplayers\/text()#",
            "type": "text",
            "position": 3
        },
        {
            "name": "Duration (mins)",
            "path": "#\/\/playingtime\/text()#",
            "type": "text",
            "position": 4
        },
        {
            "name": "Publisher",
            "path": "#(\/\/boardgamepublisher)[1]\/text()#",
            "type": "text",
            "position": 5
        },
        {
            "name": "Creator",
            "path": "#\/\/boardgamedesigner\/text()#",
            "type": "text",
            "position": 6
        }
    ]
}
```

**Comment by Gangstrocity** [comment]
Hello and thanks for this. I am unable to get it working with these settings. Would you mind taking a look and let me know if you can see what I need to change?


![image](https://github.com/user-attachments/assets/862d3ecc-e6c0-4b54-a310-c0d4be958c54)

**Reply by frank20a** [reply]
I haven't used this in a while and I do not currently have koillection installed on my setup. So it is difficult for me to check unfortunately. I plan to reinstall it after Christmas vacations in the end of January though if you can wait a bit...

**Reply by Gangstrocity** [reply]
Hey man I just got around to trying this again and it worked perfectly this time. Copied your code into a .json file and uploaded it as a new scraper. Followed the instructions and it works. Thank you!
